src/apps/vpn/appconstants.cpp

```diff
@@ -19,7 +19,15 @@ const QString& AppConstants::getStagingServerAddress() {
 
 void AppConstants::setStaging() {
   Constants::setStaging();
+
+  // Get staging server address. If it was set to empty by the user, remove it
+  // from SettingsHolder, and then query for it again to get the default
+  // address.
   s_stagingServerAddress = SettingsHolder::instance()->stagingServerAddress();
+  if (s_stagingServerAddress.isEmpty()) {
+    SettingsHolder::instance()->removeStagingServerAddress();
+    s_stagingServerAddress = SettingsHolder::instance()->stagingServerAddress();
+  }
```

Comment:
That way, if we are on production and want to switch back to staging, the address will already be there and we don't have to hunt for it

Reply:
There are multiple calls to AppConstants::setStaging() in the code base, so I'd prefer not to make changes in all those call sites. I thought it is not possible for a user to change the server address when not staging?

Reply:
Ahhh, I see now - since we have to restart the app when we go from production -> staging anyways, there is no need to restore the URL when we are not on staging (eg deleting the address, then toggling staging off). This looks good to me 👍